### DISCUSSION TITLE
Accepting other mouse models

### DISCUSSION THREAD

**Discussion by YeaSeb** [YeaSeb/wmo-oc-kmod#3] [General]
Finally, I'm able to overclock my WMO1.1a (045e:0040 in lsusb) again after bulding my first desktop in 8 years and finding out some modern motherboards don't really have a direct EHCI controller (everything goes through XHCI, even with USB2.0 ports). gcadapter_oc from HannesMann it's a godsend and I've taken upon myself the task to fork it for mouses, as this module doesn't even care about USB hubs, making it possible to use it with anything and everything AFAIK.

So I've been asking myself where to go from here for a while now, and I know that 045e:0040 may not be the only mouse with a MLT04, and that sensor may not be the only one that overclocks. For now, to try this trick with another mouse you need to edit lines 6 and 7 of wmo_oc.c and recompile the module, which isn't ideal. 

So, for you reading this, for me and for everyone using this module, what would you think is a better option?

Do we start accepting request to manually add new VID:PID combinations as some data structure in the code itself? (would bloat the code and make development kinda slow)

Do we edit the code to accept new VID:PID combinations as kernel parameters? (This would probably be a good option, but by that time this would be a usb_pollrate_oc making it possible for end users to break something) 

Or do we try to figure out usbhid so it can overclock any mouse identified by it? (Which could potentially break things with no intervention of the user other than installing the module)

I am eager to hear back from you all.

**Comment by nmw** [comment]
I think it would probably be good to pull any "overclockable" devices in with the option to overclock them.  if there's a way to get a tree in /sys/module/wmo_oc for each device it sees with its own rate parameter, with 0 for ignore (don't automatically reset polling rate), it might work well.

one of the issues with the old usbhid.conf, etc, was that it just has one parameter for mouse, one for keyboard, and one for joystick.  (none of which seem to do anything on XHCI).  Some more complicated setups could have more than one of each type of device.  for example: I have a trackball that I use for general work, and a gaming mouse for gaming.  The trackball doesn't work properly at higher polling rates, so it would be good to have a separate knob for each device.

Also, I'm curious about other devices.  Gamepads, specifically.  I found this when trying to figure out a way to get a PS5 controller working at >250Hz.   It's trivial in windows, but it seems nobody knows how to do it in linux, (and most don't understand why you would want to)
